packages/react-input/src/components/Input/Input.types.ts

```diff
@@ -79,10 +76,35 @@ export type InputProps = Omit<
    * Called when the user changes the input's value.
    */
   onChange?: (ev: React.FormEvent<HTMLInputElement>, data: InputOnChangeData) => void;
+
+  /**
+   * An input can have different text-based [types](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#input_types)
+   * based on the type of value the user will enter.
+   *
+   * Note that no custom styling is currently applied for alternative types, and some types may
```

Comment:
I like this choice and this is a good description of the implications for users.